examples/mlx_metal_kernel_opt/evaluator.py

```
@@ -0,0 +1,1395 @@
+                            f"    [{i}/{len(custom_configs)}] Command-buffer-protected: {config.name}"
+                        )
+
+                        benchmark_retry = 0
+                        while benchmark_retry <= 2:  # Fewer retries per benchmark
+                            try:
+                                # Clean state before each benchmark
+                                self._ensure_clean_gpu_state()
+
+                                # Run with maximum protection
+                                success, result = self._bulletproof_execute_with_gpu_protection(
+                                    lambda: self.benchmark_suite.run_single_benchmark(config)
```

Ruff (0.11.9)

Function definition does not bind loop variable `config`

(B023)